Services/AdminService.cs

```
@@ -0,0 +1,112 @@
+    }
+    public async Task<User> ModifyUser(string id, CreateUserDto newValues)
+    {
+        var user = await GetUser(id);
+        user.UserName = newValues.Username;
+        user.Email = newValues.Email;
+        user.Name = newValues.Name;
+        user.LastName = newValues.LastName;
+        user.Email = newValues.Email;
+        var resultModifyUser = await _userManager.UpdateAsync(user);
+        if (!resultModifyUser.Succeeded)
+        {
+            var formattedErrors = IdentityErrorHelper.FormatErrors(resultModifyUser.Errors);
+            throw new IdentityValidationEx(formattedErrors);
+        }
+
+        var token = await _userManager.GeneratePasswordResetTokenAsync(user);
+        var resultResetPassword = await _userManager.ResetPasswordAsync(user, token, newValues.Password);
+
+        if (!resultResetPassword.Succeeded)
+        {
+            var formattedErrors = IdentityErrorHelper.FormatErrors(resultResetPassword.Errors);
+            throw new IdentityValidationEx(formattedErrors);
+        }
+        var roles = await _userManager.GetRolesAsync(user);
+        if (roles[0] != newValues.Role)
+        {
+            var resultRemoveRole = await _userManager.RemoveFromRoleAsync(user, roles[0]);
+            if (!resultRemoveRole.Succeeded)
+            {
+                var formattedErrors = IdentityErrorHelper.FormatErrors(resultRemoveRole.Errors);
+                throw new IdentityValidationEx(formattedErrors);
+            }
+            var resultAddRole = await _userManager.AddToRoleAsync(user, newValues.Role);
+            if (!resultAddRole.Succeeded)
+            {
+                var formattedErrors = IdentityErrorHelper.FormatErrors(resultAddRole.Errors);
+                throw new IdentityValidationEx(formattedErrors);
+            }
+        }
+        return user;
+    }
+
+    public async Task DeleteUser(string id)
+    {
+        var user = await GetUser(id);
+        await _userManager.DeleteAsync(user);
```

What if user is null?

---

The GetUser method contemplates that if user == null a corresponding error will be thrown

 public async Task<User> GetUser(string id)
    {
        return await _userManager.Users.FirstOrDefaultAsync(u => u.Id == id) ?? throw new Exception("User not found");
    }